File: assignment 10 - C# practice problems/22.cs
//Write a c# function to test whether a date is a weekend.

using System;
namespace MyNameSpace
{
    class Program22
    {
        public static void TestWeekendDriverFunction()
        {


            Console.WriteLine("Enter 2 dates in dd/mm/yyyy format");
            var dateInput1 = Console.ReadLine();
            var split1 = dateInput1.Split('/');

            var date1Obj1 = new DateTime(int.Parse(split1[2]), int.Parse(split1[1]), int.Parse(split1[0]), 12, 0, 0);

            var day = (int)date1Obj1.DayOfWeek;
            if (day == 0 || day == 6)
            {
                Console.WriteLine("Date is a weekend");
            }
            else
            {
                Console.WriteLine("Date is not a weekend");
            }
        }
    }
}

File: assignment 10 - C# practice problems/js/22.js
//Write a JavaScript function to test whether a date is a weekend.
function testWeekend(dateParam) {

    let dateObj = new Date(dateParam[2], dateParam[1] - 1, dateParam[0]); //january is 0 feb is 1

    let day = dateObj.getDay();
    if (day === 0 || day === 6) {
        alert("Date is a weekend");
    } else {
        alert("Date is not a weekend");
    }
}

function testWeekendDriverFunction() {
    let dateInput = prompt("Enter date in dd/mm/yyyy");

    testWeekend(dateInput.split("/"));
}

testWeekendDriverFunction();

//sunday is zero
//saturday is six